### ISSUE TITLE
jraoul2002: Asynchronous Programming

### ISSUE THREAD

**Issue by jraoul2002** [HackYourFutureBelgium/class-11-12#287]
> [module repository](https://github.com/HackYourFutureBelgium/asynchronous-programming)

## Suggested Study

1- The Event Loop
      * [Loupe](http://latentflip.com/loupe/) (+10)
      * [In the Loop](https://www.youtube.com/watch?v=cCOL7MC4Pl0) (+10)
      * `setTimeout and setInterval:` [js.info](https://javascript.info/settimeout-setinterval), [Beau](https://www.youtube.com/watch?v=kOcFZV3c75I)
2-Isolate
      * Callstack
      * Closure (1. identifying closure)
      * Labeled Logger

---

## Week 1

- [ ] Sunday Prep

- [ ] Sunday Class

### Project : Pomofocus
- [x] [repo](https://github.com/jraoul2002/Pomofocus) (with a complete README)
- [x] [live demo](https://jraoul2002.github.io/Pomofocus/)
- Planning
  - [ ] [Backlog](https://github.com/_/_/tree/master/project-planning/backlog.md)
  - [ ] [Development Strategy](https://github.com/_/_/tree/master/project-planning/development-strategy.md)
- Implementation
  - [ ] ES Modules (`import`/`export`)
  - [ ] at least one `class`
  - [ ] at least one `setTimeout` and one `setInterval`
  - [ ] Logs of each user interaction
-->

### Check-In

__Progress__
done with some exercises
__Blocked__

__Next Steps__
starting to work on the week project.
__Tip(s) of the week__

**Comment by ThibaultLesuisse** [comment]
I don't think you are ready yet but I'll give you some tips to get the job done:

1. Try to uses classes in your solution. Try to break your code into smaller pieces. For example you could create a `Task` and `Timer` class. 
2. At the moment the majority of you code resides in the `index.html` file. Try and link the `init.js file in your `index.html`.
2. You don't have a start button, try to add one
3. Your stop button resets the timer. On [https://pomofocus.io/](pomofocus) clicking the stop button pauses the timer.
4. If you still have some time left try and add the tasks which act like a todo list (preferably in a seperate class)

**Comment by bramdevries** [comment]
About your portfolio assignment:

- You did a good job splitting up the work into small chunks such as starting with the about page, then the projects and finally the contact page. I also like that you small commits to add these instead of committing everything at once.
- In your [`init.js`](https://github.com/jraoul2002/Portfolio/blob/master/src/init.js#L11) the event listener is not correctly configured to wait for the `onload` event to occur because you're already executing the `appHandler` function. Functionality-wise this doesn't make much of a difference but it's something to be careful about.
- Good job on the fetching of the user data and repositories with the GitHub API. When you fetch the repositories you're also doing some sorting. The [GitHub API](https://docs.github.com/en/rest/reference/repos#list-repositories-for-a-user) also allows you to choose in which order they are returned using the `sort` query parameter, so you could refactor this to use that.
- There are some repositories where for example a homepage is not available but the component still shows this. Instead of rendering something empty you could also choose to not display the `Homepage` title. There's a similar thing happening with the bio but in this case it displays `NULL`.
